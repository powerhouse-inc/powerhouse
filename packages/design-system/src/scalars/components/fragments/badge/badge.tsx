import React from "react";
import { cva, type VariantProps } from "class-variance-authority";
import { cn } from "@/scalars/lib/utils";

const badgeVariants = cva(
  cn(
    "inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold",
    "transition-colors focus:outline-none",
  ),
  {
    variants: {
      variant: {
        default: "border-transparent",
      },
    },
    defaultVariants: {
      variant: "default",
    },
  },
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {}

const Badge: React.FC<BadgeProps> = ({ variant, className, ...props }) => (
  <div className={cn(badgeVariants({ variant }), className)} {...props} />
);

export { Badge, badgeVariants };
