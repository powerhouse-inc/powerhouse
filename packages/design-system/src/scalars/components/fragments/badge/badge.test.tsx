import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Badge } from "./badge";

describe("Badge Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<Badge />);
    expect(asFragment()).toMatchSnapshot();
  });
});
