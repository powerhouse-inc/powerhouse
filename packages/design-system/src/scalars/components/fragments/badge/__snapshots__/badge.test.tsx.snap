// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none border-transparent"
  />
</DocumentFragment>
`;
